docs/src/details/main-modules/draw/draw_descriptors.rst

```diff
@@ -0,0 +1,706 @@
+descriptor.
+
+.. lv_example:: grad/lv_example_grad_2
+  :language: c
+
+
+Radial Gradients
+----------------
+
+The radial gradient is described by two circles: an outer circle and an inner circle
+(also called the focal point).  The gradient will be calculated between the focal
+point's circle and the edge of the outer circle.
+
+If the center of the focal point and the center of the main circle are the same, the
+gradient will spread evenly in all directions.  If the center points are not the
+same, the gradient will have an egg shape.
+
+The focal point's circle should be inside the main circle.
+
+After initializing the stops with :cpp:func:`lv_grad_init_stops`, the outer
+circle can be set by:
+:cpp:expr:`lv_grad_radial_init(&grad_dsc, center_x, center_y, edge_x, edge_y, LV_GRAD_EXTEND_PAD_...)`
+
+For both the center and edge coordinates, ``px`` or ``lv_pct()`` values can be used.
+
+The inner circle (focal point) can be set with:
+:cpp:expr:`lv_grad_radial_set_focal(&grad_dsc, center_x, center_y, radius)`
+
+.. lv_example:: grad/lv_example_grad_3
+  :language: c
+
+
+Conic Gradients
+---------------
+
+The conic gradient is defined between the angles of a circle, and colors are mapped
+to each angle.
+
+After initializing the stops with :cpp:func:`lv_grad_init_stops`, the conic gradient
+can be set up with:
+:cpp:expr:`lv_grad_conical_init(&grad, center_x, center_y, angle_start, angle_end, LV_GRAD_EXTEND_PAD_...)`
+
+For both the center and edge coordinates, ``px`` or ``lv_pct()`` values can be used.
+
+The zero angle is on the right-hand side, and 90 degrees is at the bottom.
+
+.. lv_example:: grad/lv_example_grad_4
+  :language: c
+
+
+
+Border Draw Descriptor
+**********************
+
+The :cpp:type:`lv_draw_border_dsc_t` border descriptor has radius, opacity,
+width, color, and side fields.  If the opacity or width is 0, no Draw Task will
+be created.
+
+``side`` can contain ORed values of :cpp:type:`lv_border_side_t`, such as
+:cpp:enumerator:`LV_BORDER_SIDE_BOTTOM`.  :cpp:enumerator:`LV_BORDER_SIDE_ALL`
+applies to all sides, while :cpp:enumerator:`LV_BORDER_SIDE_INTERNAL` is used by
+higher layers (e.g. a table Widget) to calculate border sides.  However, the drawing
+routine receives only simpler values.
+
+The following functions are used for border drawing:
+
+- :cpp:expr:`lv_draw_border_dsc_init(&dsc)` initializes a border Draw Task.
+- :cpp:expr:`lv_draw_sw_border(layer, &dsc, area)` creates a Draw Task to draw a border
+  inward from its area.
+- :cpp:expr:`lv_draw_task_get_border_dsc(draw_task)` retrieves the border descriptor
+  from a Draw Task.
+
+.. lv_example:: styles/lv_example_style_3
+  :language: c
+
+
+Outlines
+********
+
+The outline is similar to the border but is drawn outside the object's draw area.
+
+In practice, there is no dedicated outline descriptor like
+:cpp:type:`lv_draw_outline_dsc_t`, because from the rendering perspective, the
```

Comment:
I've just noticed it. Once we say it doesn't exist, we shouldn't try to create a link to it either. 
```suggestion
``lv_draw_outline_dsc_t``, because from the rendering perspective, the
```